datafusion/common/src/utils.rs

```diff
@@ -78,6 +78,29 @@ pub fn project_schema(
     Ok(schema)
 }
 
+/// Checks if the given projection is valid for the given schema.
+pub fn can_project(schema: &SchemaRef, projection: Option<&Vec<usize>>) -> Result<()> {
```

Comment:
I'm not sure if this is enough common to be put here (`common`)? Seems you can keep it in the module it is used for now. If we have more usage of it, we can pull it to more common place later. So we don't increase public APIs unnecessarily.

Reply:
> So we don't increase public APIs unnecessarily.

Agree with it. I place it here because I think it can be used in #7932 . Shall we create `utils.rs` under  `datafusion/physical-plan/src/analyze.rs` ?

Reply:
There is already `datafusion/physical-plan/src/common.rs`, maybe?

Reply:
> There is already `datafusion/physical-plan/src/common.rs`, maybe?

I will move it there.